src/compiler/gcc.rs

```diff
@@ -311,14 +322,22 @@ pub fn preprocess<T>(creator: &T,
     where T: CommandCreatorSync
 {
     trace!("preprocess");
+    let language = match parsed_args.language {
+        Language::C => "c",
+        Language::Cxx => "c++",
+        Language::ObjectiveC => "objective-c",
+        Language::ObjectiveCxx => "objective-c++",
```

Comment:
Sort of a bummer that there are so many ways to spell these that you can't just have a common method on `Language`.